OpenRA.Mods.Common/Widgets/UnitCommandWidget.cs

```diff
@@ -0,0 +1,178 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Orders;
+using OpenRA.Primitives;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	/// <summary> Contains all functions that are unit-specific. </summary>
```

Comment:
A more accurate comment would be

```csharp
/// <summary>A giant collection of hardcoded hacks because we still don't have a way for traits to register hotkeys</summary>
```